odoo/service/model.py

```diff
@@ -106,7 +107,18 @@ def _(src):
                 # Automatically retry the typical transaction serialization errors
                 if e.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
                     raise
-                if tries >= MAX_TRIES_ON_CONCURRENCY_FAILURE:
+                if max_tries is None:
+                    # TODO: refactor to reuse code from tr function above
+                    try:
+                        from odoo.http import request
+                        ctx = request.env.context
+                        max_tries = ctx.get("MAX_TRIES_ON_CONCURRENCY_FAILURE")
+                    except Exception:
+                        pass
+                    if max_tries is None:
+                        max_tries = MAX_TRIES_ON_CONCURRENCY_FAILURE
+
+                if tries >= max_tries:
```

Comment:
IMHO this is useless, as long as the concurrency error is caught and properly ignored by the callee.

Reply:
I think that without this, it will only raise **after** the `MAX_TRIES_ON_CONCURRENCY_FAILURE` is reached, which is what he wants to avoid

Reply:
> I think that without this, it will only raise **after** the `MAX_TRIES_ON_CONCURRENCY_FAILURE` is reached, which is what he wants to avoid

yes, it's my understanding of how it works. I'll check again

Reply:
Apparently, system catches the error and retries only on exit from the method, while inside the method it raise immediately and can be handled